### PR TITLE
snapstate: remove temporary snap file for local revisions early

### PR DESCRIPTION
**Pull request by mvo5**
The SnapSetup.SnapPath is only needed to check/mount the snap. If
the snap is already local we will not mount it again. So we can
remove temporary snap files from e.g. side-loading early.

This fixes the bug that sideloaded snaps that already have local
revisions are not removed. They will get removed eventually via
the snapmgr.go:SnapManager.localInstallCleanup() code. But this
code only runs every 24h and if people install enough side-loaded
snaps the disk may fill-up before the cleanup happens.

### CODE REVIEW

File: overlord/snapstate/snapstate.go [hunk 1]
```diff
@@ -287,6 +287,19 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	// check if we already have the revision locally (alters tasks)
 	revisionIsLocal := snapst.LastIndex(targetRevision) >= 0
 
+	// We will not need the temporary snap if we have a local revision.
+	// This can happen when e.g. side-loading a local revision again.
+	// The SnapPath is only needed in the "mount-snap" handler and
+	// that is skipped for local revisions.
+	//
+	// XXX: should this be closer to where we skip generating the mount
+	//      task?
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go:296]: naively I would say so but I haven't looked a the code

Comment (pedronis) [reply overlord/snapstate/snapstate.go:296]: I looked at the code, there is not much code in between, so it doesn't matter much.  I would think putting it closer to the mount task is still a win if any of this needs later changes and the extra time window looks tiny.

Comment (mvo5) [reply overlord/snapstate/snapstate.go]: Thanks, I moved it now

File: overlord/snapstate/snapstate.go [hunk 2]
```diff
@@ -323,6 +323,17 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		mount := st.NewTask("mount-snap", fmt.Sprintf(i18n.G("Mount snap %q%s"), snapsup.InstanceName(), revisionStr))
 		addTask(mount)
 		prev = mount
+	} else {
+		if snapsup.Flags.RemoveSnapPath {
+			// If the revisin is local, we will not need the
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go:328]: typo: revision

**Review by pedronis — APPROVED** [review]
lgtm, thanks, typo though

**Review by mardy — APPROVED** [review]
Great series of new tests! At first sight, they look all right :-)

**Comment by woodrow-shen** [comment]
I can confirm that issue is fixed by refreshing beta channel. Thank you all!

**Comment by mvo5** [comment]
\o/ thank you!

**Pull request merged**